### PR TITLE
[AIRFLOW-4419] Restore used_slots and queued_slots Pool methods

### PR DESCRIPTION
**Pull request by ashb**
Make sure you have checked _all_ steps below.

### Jira

- [x] https://issues.apache.org/jira/browse/AIRFLOW-4419

### Description

- [x] These were removed in #5194 but were still used in the UI and are helpful for observability of Ariflow, so I've added them back.

  /cc @feng-tao @KevinYang21

### Tests

- [x] Added a simple test to the PoolModelView to ensure the expected links are there.

  I haven't tested the behaviour of the used/queued slots as they are simple enough functions

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/models/pool.py [hunk 1]
```diff
@@ -64,6 +95,8 @@ def open_slots(self, session):
         from airflow.models.taskinstance import \
             TaskInstance as TI  # Avoid circular import
 
+        # Issue a single query instead of using the used_slots/queued_slots to
+        # avoid load on DB
         used_slots = session.query(func.count()).filter(TI.pool == self.pool).filter(
             TI.state.in_([State.RUNNING, State.QUEUED])).scalar()
         return self.slots - used_slots
```
Comment (BasPH) [inline airflow/models/pool.py:102]: Before the [change removing this code](https://github.com/apache/airflow/commit/d63b6c9a5d3f8dfcb959daf93dd55dcb710d0e38), the open slots was calculated as slots - used_slots - queued_slots. Should we also restore it like that?

Comment (feng-tao) [reply airflow/models/pool.py:102]: used_slot already include both tasks in running state and queued state. So no need to minus the queued slots.

**Review by feng-tao — APPROVED** [review]

File: airflow/models/pool.py (hunk 1 above)
Comment (ashb) [reply airflow/models/pool.py:102]: I think it still is? used_slots was State.RUNNING, and queued_slots was State.QUEUED.

So now its `slots - (count RUNNING or QUEUED)` = `slots - RUNNING - QUEUED` (just via one query than two).

I think. It's been a long day.

**Comment by KevinYang21** [comment]
Sry somehow my IDE didn't give me the usage in the UI. About the behavior, since in scheduler we consider both `RUNNING` and `QUEUED` as used, do you think it makes sense to just show `used slots` in the UI so it is less confusing?

**Comment by ashb** [comment]
@KevinYang21 Yeah, likely cos they are "protected" behind `.get('queued_slots')` not directly via `pool.queued_slots`. I only noticed cos I cherry-picked it to v1-10 and that failed it's tests :D

**Comment by ashb** [comment]
Changing the UI is probably a different issue/ticket.

**Pull request merged**